pipeline/scripts/ci/upstream_cli.py

```diff
@@ -0,0 +1,160 @@
+"""
+This script provides the upstream latest rpm repo path and container image.
+Please note that this script should be run under root privilages.
+and podman should be installed on that host.
+"""
+import logging
+import os
+import re
+import sys
+
+import requests
+import yaml
+from docopt import docopt
+
+LOG = logging.getLogger(__name__)
+
+
+def exec_sys_commands(cmdlist):
+    """Method to execute system related commands.
+    Args:
+        cmdlist: List of commands to execute.
+    """
+    for cmd in cmdlist:
+        exit_status = os.system(cmd)
+        if exit_status:
+            raise Exception(f"Command failed to execute {cmd}")
+
+
+def store_in_upstream(branch_name, repo, image, version):
+    """Method to update it in upstream.yaml.
+    Args:
+        branch_name: Name of upstream branch
+        repo: upstream branch compose URL
+        image: upstream branch image
+        version: upstream branch ceph version.
+    """
+    url = "http://magna002.ceph.redhat.com/cephci-jenkins/latest-rhceph-container-info/upstream.yaml"
+    data = requests.get(url, timeout=30)
+    repo_details = yaml.safe_load(data.text)
+    existing_version = repo_details[branch_name]["ceph-version"]
```

Comment:
This might fail if it is first time update to the file.

Reply:
yes handled that condition and raised exception if key i.e branch name not found.